### PR TITLE
[docs] Add docs for non-SLURM cluster setup

### LINKED ISSUE #5746: [docs] Add documentation for non-slurm computing cluster setup
Make the documentation for the functionality of #1387

See the discussion at #1345

### PR DESCRIPTION
**Pull request by areshytko**
## What does this PR do?

Fixes #5746

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [ ] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified
 - [x] **Check that target branch and milestone match!**

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -96,6 +96,7 @@ PyTorch Lightning Documentation
    cloud_training
    amp
    slurm
+   cluster
    child_modules
```
Comment (awaelchli) [inline docs/source/index.rst:100]: just an idea, we could combine the slurm and cluster navigation items into one and make instead two sections, one for slurm and one for non-slurm

File: docs/source/cluster.rst [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+Also, check :ref:`accelerators` as a new and more general approach to a cluster setup.
+
+--------
+
+
+Cluster setup
+-------------
+
+To setup multi-node computing cluster you need:
```
Comment (awaelchli) [inline docs/source/cluster.rst:21]:
```suggestion
To setup a multi-node computing cluster you need:
```
is there an article missing here?

File: docs/source/cluster.rst [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+PyTorch Lightning follows the design of `PyTorch distributed communication package <https://pytorch.org/docs/stable/distributed.html#environment-variable-initialization>`_. and requires the following environment variables to be defined on each node:
+
+- *MASTER_PORT* - required; has to be a free port on machine with NODE_RANK 0
+- *MASTER_ADDR* - required (except for NODE_RANK 0); address of NODE_RANK 0 node
+- *WORLD_SIZE* - required; how many nodes are in the cluster
+- *NODE_RANK* - required; id of the node in the cluster
+
+
+Train script design
+-------------------
```
Comment (awaelchli) [inline docs/source/cluster.rst:36]:
```suggestion
Training script design
----------------------
```

File: docs/source/cluster.rst [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+To train a model using multiple nodes, do the following:
+
+1.  Design your :ref:`lightning_module` (no need to add anything specific here).
+
+2.  Enable DDP in the trainer
+
+    .. code-block:: python
+
+       # train on 32 GPUs across 4 nodes
+       trainer = Trainer(gpus=8, num_nodes=4, distributed_backend='ddp')
```
Comment (awaelchli) [inline docs/source/cluster.rst:47]:
```suggestion
       trainer = Trainer(gpus=8, num_nodes=4, accelerator='ddp')
```

File: docs/source/cluster.rst [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+
+Submit a job to the cluster
+---------------------------
+
+To submit a training job to the cluster you need to run the same training script on each node of the cluster.
+This means that you need to:
+
+1. Copy all third-party libraries to each node (usually means - distribute requirements.txt file and install it).
+
+2. Copy all your import dependencies and script itself to each node.
```
Comment (awaelchli) [inline docs/source/cluster.rst:58]:
```suggestion
2. Copy all your import dependencies and the script itself to each node.
```

**Review by awaelchli — APPROVED** [review]
I love these docs!

File: docs/source/cluster.rst [hunk 6]
```diff
@@ -0,0 +1,60 @@
+.. testsetup:: *
+
+    from pytorch_lightning.trainer.trainer import Trainer
```
Comment (Borda) [inline docs/source/cluster.rst:3]:
```suggestion
```
it shall be already in conf.py and not rather add it there :]

**Review by Borda — COMMENTED** [review]
I would rather see this change in feat 1.2

**Review by justusschock — APPROVED** [review]

**Review by ydcjeff — APPROVED** [review]

**Review by s-rog — APPROVED** [review]
Very cool, will try this myself at some point!

File: docs/source/cluster.rst [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+
+
+Cluster setup
+-------------
+
+To setup a multi-node computing cluster you need:
+
+1) Multiple computers with PyTorch Lightning installed
+2) A network connectivity between them with firewall rules that allow traffic flow on a specified *MASTER_PORT*.
+3) Defined environment variables on each node required for the PyTorch Lightning mutli-node distributed training
```
Comment (rohitgr7) [inline docs/source/cluster.rst:22]:
```suggestion
To setup a multi-node computing cluster you need:

1) Multiple computers with PyTorch Lightning installed
2) A network connectivity between them with firewall rules that allow traffic flow on a specified *MASTER_PORT*.
3) Define environment variables on each node required for the PyTorch Lightning multi-node distributed training
```

**Review by rohitgr7 — APPROVED** [review]

File: docs/source/cluster.rst (hunk 7 above)
Comment (s-rog) [reply docs/source/cluster.rst:22]: I think the "Defined" is correct
